object-common.go

```diff
@@ -71,7 +71,7 @@ func cleanupDir(storage StorageAPI, volume, dirPath string) error {
 		}
 		return nil
 	}
-	return delFunc(retainSlash(dirPath))
+	return delFunc(pathJoin(dirPath))
```

Comment:
it should be retainSlash() here because for delFunc() if the argument has a trailing slash it means it is a directory